App crashes after downloading certain videos

### Steps to reproduce

Example:
1. Download this video https://www.youtube.com/watch?v=k6jqx9kZgPM
2. Select your preferred video and audio quality
3. Press download
4. App crashes

### Expected behavior

The video will be downloaded and play successfully.

### Actual behavior

App crashes and shows error log after reopening.

### LibreTube version

0.7.0

### Android version

Android 8.1

### Other details

![Screenshot_2022_1030_111857](https://user-images.githubusercontent.com/106541892/198861020-f0b4ca2f-f614-4017-983d-30b0146c5fb7.png)
This shows after crashing and reopening.

Full log:
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:875)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
	... 1 more
Caused by: java.io.IOException: No such file or directory
	at java.io.UnixFileSystem.createFileExclusively0(Native Method)
	at java.io.UnixFileSystem.createFileExclusively(UnixFileSystem.java:281)
	at java.io.File.createNewFile(File.java:1000)
	at com.github.libretube.ui.dialogs.DownloadDialog$$ExternalSyntheticLambda2.onClick(R8$$SyntheticClass:207)
	at android.view.View.performClick(View.java:6325)
	at com.google.android.material.button.MaterialButton.performClick(MaterialButton.java:10)
	at android.view.View$PerformClick.run(View.java:25114)
	at android.os.Handler.handleCallback(Handler.java:790)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:192)
	at android.app.ActivityThread.main(ActivityThread.java:6759)
	... 3 more
    

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/extensions/Sanitize.kt
+++ b/app/src/main/java/com/github/libretube/extensions/Sanitize.kt
@@ -4,5 +4,5 @@ package com.github.libretube.extensions
  * Replace file name specific chars
  */
 fun String.sanitize(): String {
-    return this.replace("[^a-zA-Z0-9\\._]+", "_")
+    return this.replace("[^a-zA-Z0-9\\\\._]+".toRegex(), "_")
 }

--- a/app/src/main/java/com/github/libretube/ui/dialogs/DownloadDialog.kt
+++ b/app/src/main/java/com/github/libretube/ui/dialogs/DownloadDialog.kt
@@ -124,7 +124,7 @@ class DownloadDialog(
         if (binding.audioSpinner.size >= 1) binding.audioSpinner.setSelection(1)
 
         binding.download.setOnClickListener {
-            if (binding.fileName.text.toString().length < 1) {
+            if (binding.fileName.text.toString().isEmpty()) {
                 Toast.makeText(context, R.string.invalid_filename, Toast.LENGTH_SHORT).show()
                 return@setOnClickListener
             }

